src/kademlia/dht_storage.cpp

```diff
@@ -119,9 +119,18 @@ namespace
 		std::memcpy(item.value.get(), buf.data(), size);
 	}
 
+	system_clock::time_point clock_now()
+	{
+		static auto const stime = system_clock::now();
+		static auto const ctime = aux::time_now();
+
+		auto const secs = total_seconds(aux::time_now() - ctime);
+		return stime + seconds(secs);
```

Comment:
oh, you need to do this to get the simulations to work, right? maybe an easier way to do this would be to use the internal clock here, and just convert to and from ``time_t`` when saving and restoring